### ISSUE TITLE
Clean-up after logout

### ISSUE THREAD

**Issue by csarven** [solid/solid-auth-client#96]
Is there a particular reason why `logout()` doesn't remove `solid-auth-client` in `localStorage`? Since the session is killed, there is nothing in particular in solid-auth-client in localStorage that's useful (AFAICT). `requestHistory` appears to grow on subsequent logins with solid-auth-client sticking around. Hmm. Perhaps do a complete clean-up on logout()?

**Comment by RubenVerborgh** [comment]
> Is there a particular reason why `logout()` doesn't remove `solid-auth-client` in `localStorage`?

Your IDP is remembered for next time, very important for usability.

> `requestHistory` appears to grow on subsequent logins with solid-auth-client sticking around.

Yes, we probably want to remove that one, or at least prevent arbitrary growth.

**Comment by csarven** [comment]
Do you mean that it is remembered so that popup should be able to reuse it? Auto-sign-in when it comes up? If that's so, I'm not experiencing that ie. I'll have to select the IdP again.

**Comment by csarven** [comment]
or do you mean that the applications can reuse that information ie skipping popup?

**Comment by RubenVerborgh** [comment]
> Do you mean that it is remembered so that popup should be able to reuse it?

Yes. Especially visible in 2.2.13 now (not hidden behind "custom IDP" button anymore).

> Auto-sign-in when it comes up?

That was deemed undesirable because then you couldn't switch accounts on the same IDP.

> or do you mean that the applications can reuse that information ie skipping popup?

Not what I meant, but actually possible too!

**Comment by scenaristeur** [comment]
I think localstorage should be cleaned, because we cannot switch from one account to another. with let popupUri = 'https://solid.community/common/popup.html';
